pointgrey_camera_driver/src/nodelet.cpp

```diff
@@ -59,22 +59,24 @@ class PointGreyCameraNodelet: public nodelet::Nodelet
 
   ~PointGreyCameraNodelet()
   {
+    boost::mutex::scoped_lock scopedLock(connect_mutex_);
+
     if(pubThread_)
     {
       pubThread_->interrupt();
       pubThread_->join();
-    }
 
-    try
```

Comment:
I don't think we need to stop + disconnect if the thread was already destroyed or never spawned.